src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java

```diff
@@ -65,47 +65,60 @@ public class GroupNodeViewModel {
     private final BooleanBinding allSelectedEntriesMatched;
     private final TaskExecutor taskExecutor;
     private final CustomLocalDragboard localDragBoard;
-    private final ObservableList<BibEntry> entriesList;
     private final PreferencesService preferencesService;
     private final InvalidationListener onInvalidatedGroup = listener -> refreshGroup();
 
+    private final ObservableList<BibEntry> entriesList;
+
     public GroupNodeViewModel(BibDatabaseContext databaseContext, StateManager stateManager, TaskExecutor taskExecutor, GroupTreeNode groupNode, CustomLocalDragboard localDragBoard, PreferencesService preferencesService) {
         this.databaseContext = Objects.requireNonNull(databaseContext);
         this.taskExecutor = Objects.requireNonNull(taskExecutor);
         this.stateManager = Objects.requireNonNull(stateManager);
         this.groupNode = Objects.requireNonNull(groupNode);
         this.localDragBoard = Objects.requireNonNull(localDragBoard);
         this.preferencesService = preferencesService;
-
-        displayName = new LatexToUnicodeFormatter().format(groupNode.getName());
+        LatexToUnicodeFormatter formatter = new LatexToUnicodeFormatter();
+        groupNode.nameSubscription(name -> displayName = formatter.format(name));
         isRoot = groupNode.isRoot();
         if (groupNode.getGroup() instanceof AutomaticGroup automaticGroup) {
-            children = automaticGroup.createSubgroups(this.databaseContext.getDatabase().getEntries())
-                                     .stream()
-                                     .map(this::toViewModel)
-                                     .sorted((group1, group2) -> group1.getDisplayName().compareToIgnoreCase(group2.getDisplayName()))
-                                     .collect(Collectors.toCollection(FXCollections::observableArrayList));
+            children = automaticGroup
+                .createSubgroups(this.databaseContext.getDatabase().getEntries())
+                .stream()
+                .map(this::toViewModel)
+                .sorted((group1, group2) -> group1.getDisplayName().compareToIgnoreCase(group2.getDisplayName()))
+                .collect(Collectors.toCollection(FXCollections::observableArrayList));
+            children.forEach(c -> groupNode.addChild(c.getGroupNode()));
         } else {
             children = EasyBind.mapBacked(groupNode.getChildren(), this::toViewModel);
         }
         if (groupNode.getGroup() instanceof TexGroup) {
             databaseContext.getMetaData().groupsBinding().addListener(new WeakInvalidationListener(onInvalidatedGroup));
         }
         hasChildren = new SimpleBooleanProperty();
-        hasChildren.bind(Bindings.isNotEmpty(children));
+        hasChildren.bind(Bindings.isNotEmpty(groupNode.getChildren()));
```

Comment:
The old code did not work?

Reply:
Think there was a potential problem with the main table context menu that this was for, but since that's been removed I will revert.